packages/filebrowser/src/listing.ts

```diff
@@ -1138,7 +1138,13 @@ export class DirListing extends Widget {
     // Write to DOM
     for (const column of visibleColumns) {
       const size = this._columnSizes[column.id];
-      column.element.style.width = size === null ? '' : size + 'px';
+      const newSize = size === null ? '' : size + 'px';
+      const varName =
+        column.className.replace(RegExp(/^jp-id-/), '--jp-filebrowser-') +
+        '-column-width';
+
+      // Set at the root.
+      document.documentElement.style.setProperty(varName, newSize);
```

Comment:
This will not work in general - there may be more than one file browser, e.g. when using dialog which include a file selector.

Also, we should not add new variables to the `:root` anyway due to performance reasons https://github.com/jupyterlab/jupyterlab/issues/16660

Reply:
Maybe we should set `grid-template-columns` directly via JS on the browser instance: on the header and on the content. This would resolve the TODO around de-duplication, remove the CSS variables, and prevent resizing columns of all filebrowsers at once.